program/tests/distribute_rewards.rs

```diff
@@ -135,16 +231,150 @@ async fn fail_holder_rewards_pool_invalid_data() {
     );
 }
 
+struct InitialPool {
+    token_supply: u64,
+    rewards_per_token: u128,
+    total_rewards: u64,
+}
+
+struct ExpectedPool {
+    rewards_per_token: u128,
+    total_rewards: u64,
+}
+
 #[allow(clippy::arithmetic_side_effects)]
+#[test_case(
+    InitialPool {
+        token_supply: 0,
+        rewards_per_token: 0,
+        total_rewards: 0,
+    },
+    ExpectedPool {
+        rewards_per_token: 0,
+        total_rewards: 100_000,
+    },
+    100_000;
+    "Zero token supply, zero rewards per token, increment total rewards"
+)]
+#[test_case(
+    InitialPool {
+        token_supply: 100_000,
+        rewards_per_token: 0,
+        total_rewards: 0,
+    },
+    ExpectedPool {
+        rewards_per_token: 2_500_000_000, // 0% + 250_000 / 100_000 = 250%
+        total_rewards: 250_000,
+    },
+    250_000;
+    "Zero initial rate and rewards, resulting rate 250%"
+)]
+#[test_case(
+    InitialPool {
+        token_supply: 1_000_000,
+        rewards_per_token: 0,
+        total_rewards: 0,
+    },
+    ExpectedPool {
+        rewards_per_token: 100_000_000, // 0% + 100_000 / 1_000_000 = 10%
+        total_rewards: 100_000,
+    },
+    100_000;
+    "Zero initial rate and rewards, resulting rate 10%"
+)]
+#[test_case(
+    InitialPool {
+        token_supply: 1_000_000,
+        rewards_per_token: 0,
+        total_rewards: 0,
+    },
+    ExpectedPool {
+        rewards_per_token: 1_000_000, // 0% + 1_000 / 1_000_000 = 0.1%
+        total_rewards: 1_000,
+    },
+    1_000;
+    "Zero initial rate and rewards, resulting rate 0.1%"
+)]
+#[test_case(
+    InitialPool {
+        token_supply: 1_000_000,
+        rewards_per_token: 0,
+        total_rewards: 0,
+    },
+    ExpectedPool {
+        rewards_per_token: 1_000, // 0 + 1 / 1_000_000 = 0.0001%
+        total_rewards: 1,
+    },
+    1;
+    "Zero initial rate and rewards, resulting rate 0.0001%"
+)]
```

Comment:
This doesn't need to be addressed with this PR, but these cases seem a bit repetitive -- what different behavior are they testing for? We mainly need the limits, ie. huge supply / small rewards, huge rewards / small supply, huge rewards / huge supply, etc. And then maybe one case in the middle.

Reply:
Just math, really! They can be replaced with prop tests.